src/app/components/growing-list/growing-list.component.ts

```diff
@@ -104,6 +104,12 @@ export class GrowingListComponent implements OnInit, OnDestroy {
         this.deletedItemsSubscription?.unsubscribe();
     }
 
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes.apiLink && changes.apiLink.currentValue) {
+            this.replaceApiLink(changes.apiLink.currentValue);
```

Comment:
Is this functionality required? If yes, then check if more code from onInit must be moved here (or into a separate function) to properly support updating the component (or at least warn when values change that do not support changes)

Reply:
The growing list component is a generic component. It might be used differently in the future. If there is an ApiLink, then it should be preferred over `rels`+`query`. If neither is available the list should be empty. If the component can react to changing inputs, then it must be absolutely clear, which changes get picked up and which inputs are considered final. (The requirements of a generic component do not depend solely on their current use, but also on their planned use!)

I suggest also handling changes to `newItemRels` in `onChanges`. This would also get rid of the bug where if only an api link is supplied the newItems listener in `onInit` will never get set up as the method returns immediately... (be sure to unsubscribe old subscriptions!)

The cleanest solution would be to handle all initializing logic that is based on inputs in `onChanges`. This makes sure that all changes will be picked up correctly whilst avoiding duplicate code. At the very least document in the component which input changes are expected to work.

Reply:
I updated the `ngOnChanges` method to "listen" on `apiLink`, `rels`, `query`, and `newItemRels` changes. `ngOnChanges` resets most properties and reinitializes them. I hope this suffices.